Implemented automatic generation of recurring departures, resolves #75, probably resolves #63

quittung opened the pull request:
These changes make it easier to create timetables when trains depart at regular intervals.

You can choose the separation between trains and generate the timetable for the entire hour. It takes the first timetable entry as a template, deletes the following entries and then generates entries that are offset from the template by a multiple of the separation value.

It looks like this:
![grafik](https://user-images.githubusercontent.com/10237613/120724161-85640400-c4d3-11eb-82a3-8bd83b4ef885.png)

There are three issues I'm aware of right now and need feedback on:
- The strings "Separation" and "Generate" need to be internationalized
- All timetable entries apart from the template are removed without warning
- I haven't put much thought into UI/UX design

quittung committed:
Implemented automatic generation of recurring departures as in #75

IncredibleHannes commented:
Hi,
This looks very interesting. I will have a look at it next week!

Gordon-Dry commented:
On an empty new schedule it does not work, one entry has to be added, then the "generate button" does its job and the first entry can be deleted again.

quittung committed:
Added more separation values

Changed the calculation of generated constraints to support non integer separation values (e.g. 2.5 min instead of 2 min)
Added 1.5, 2.5 and 7.5 minutes to list of separation values

quittung committed:
Inverted separation list

This way more common values are on the top of the drop down menu

quittung committed:
Merge branch 'IncredibleHannes:develop' into recurringDepartures

quittung commented:
Hey, just wondering if there's something I can do to help this process along...
